pkg/versioncheck/check.go

```diff
@@ -36,6 +38,45 @@ func MustCompile(constraint string) go_version.Constraints {
 
 // Compile trivially wraps go-version.NewConstraint, returning the constraint
 // and error
-func Compile(constraint string) (go_version.Constraints, error) {
-	return go_version.NewConstraint(constraint)
+func Compile(constraint string) (go_version.Range, error) {
```

Comment:
Add a mention of why we are wrapping the `go_version` library here with a reference to this PR / issue so that future Cilium contributors know why it is here.

Reply:
There's not much of a reason, just to make it consistent with the MustVersion/Version function calls.